File: gen.py
import random


def generate_input(d, n, q):
    # Generate n d-dimensional vectors
    vectors = []
    for _ in range(n):
        vector = [random.randint(-(10**7), 10**7) for _ in range(d)]
        vectors.append(vector)

    # Generate q queries
    queries = []
    for _ in range(q):
        query = [random.randint(-(10**7), 10**7) for _ in range(d)]
        queries.append(query)

    # Printing the input in the required format
    print(f"{d} {n}")

    for vector in vectors:
        print(" ".join(map(str, vector)))

    print(q)

    for query in queries:
        print(" ".join(map(str, query)))


# Dimensionality of the vectors (between 2 and 5)
d = random.randint(2, 5)
# Number of vectors (up to 100,000)
n = random.randint(1, 100000)
# Number of queries (up to 200,000)
q = random.randint(1, 200000)

generate_input(d, n, q)

File: gen.cpp
#include <iostream>
#include <vector>
#include <cstdlib>
#include <ctime>

using namespace std;

// Function to generate and print a random input
void generateInput(int n, int m)
{
    // Generate initial sequence of notes
    cout << n << " " << m << endl; // First line: n and m
    for (int i = 0; i < n; i++)
    {
        cout << 1 << " "; // Random note value [0, 10^9]
    }
    cout << endl;

    // Generate m operations
    for (int i = 0; i < m; i++)
    {
        int operationType = (rand() % 2) + 1; // Randomly choose 1 or 2
        int q = rand() % (1000) + 1;          // Random number of notes to add or check (1 to 1000)

        if (operationType == 1)
        {                                      // Create operation
            cout << operationType << " " << q; // Operation type and q
            for (int j = 0; j < q; j++)
            {
                cout << " " << 1; // Random non-negative note values
            }
            cout << endl;
        }
        else
        {                                      // Test album operation
            cout << operationType << " " << q; // Operation type and q
            for (int j = 0; j < q; j++)
            {
                // Pick a random length that does not exceed the current melody length
                cout << " " << (rand() % (n + i * 1000) + 1); // Ensure it doesn't exceed current length
            }
            cout << endl;
        }
    }
}

int main()
{
    // Seed the random number generator
    srand(time(0));

    int n = rand() % 5000 + 1; // Generate n in range [1, 500000]
    int m = rand() % 5000 + 1; // Generate m in range [1, 500000]

    generateInput(n, m);
    return 0;
}
